server/vulnerabilities/nvd/tools/cvefeed/nvd/match_cpe.go

```diff
@@ -69,13 +69,42 @@ func (cm *cpeMatch) Match(attrs []*wfn.Attributes, requireVersion bool) (matches
 		if cm.match(attr, requireVersion) {
 			matches = append(matches, attr)
 		}
-		if osMatch := cm.MatchTargetSW(attr); osMatch != nil {
+		if osMatch := cm.matchTargetSW(attr); osMatch != nil {
 			matches = append(matches, osMatch)
 		}
 	}
 	return matches
 }
 
+// matchTargetSW returns an OS CPE for the given application CPE
+// if the application CPE has a target software attribute and
+// matches the given cpeMatch
+func (cm *cpeMatch) matchTargetSW(attr *wfn.Attributes) *wfn.Attributes {
+	if cm == nil || attr == nil {
+		return nil
+	}
+
+	if attr.Part != "a" || attr.TargetSW == "" {
```

Comment:
Can you get stuff that's `attr.Part != "a"`, here?

Reply:
yes, you could get get an OS CPE (cpe:2.3:o:apple:macos:14.1.2:*:*:*:*:*:*:*) so we want to make sure and skip those just in case.